chain/src/Pos/Chain/Ssc/Mem.hs

```diff
@@ -48,23 +52,27 @@ askSscMem = view (lensOf @SscMemTag)
 -- | Applies state changes to given var.
 syncingStateWith
     :: TVar s
-    -> StateT s (NamedPureLogger STM) a
-    -> NamedPureLogger STM a
+    -> StateT s STM a
+    -> STM a
 syncingStateWith var action = do
-    oldV <- lift $ readTVar var
+    oldV <- readTVar var
     (res, newV) <- runStateT action oldV
-    lift $ writeTVar var newV
+    writeTVar var newV
     return res
 
 ----------------------------------------------------------------------------
 -- Local
 ----------------------------------------------------------------------------
 
-type SscLocalQuery a =
-    forall m . (MonadReader SscLocalData m, WithLogger m) => m a
+type SscLocalQuery a = forall m . Monad m =>
+    TraceNamed m ->
+    ReaderT SscLocalData m a
```

Comment:
This is going to reverse the current convention of putting the logging argument first. If we're going to establish a convention of passing the logger manually, I would *really* prefer it remain as the first argument, if at all possible.